src/v2/providers/eventarc.ts

```diff
@@ -0,0 +1,116 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import * as options from '../options';
+import { CloudEvent, CloudFunction } from '../core';
+import { copyIfPresent } from '../../common/encoding';
+import { ManifestEndpoint } from '../../runtime/manifest';
+
+/** Options that can be set on an Eventarc trigger. */
+export interface EventarcTriggerOptions extends options.EventHandlerOptions {
+  /**
+   * Type of the event.
+   */
+  eventType: string;
+
+  /**
+   * ID of the channel. Can be either:
+   *   * fully qualified channel resource name:
+   *     `projects/{project}/locations/{location}/channels/{channel-id}`
+   *   * partial resource name with location and channel ID, in which case
+   *     the runtime project ID of the function will be used:
+   *     `locations/{location}/channels/{channel-id}`
+   *   * partial channel ID, in which case the runtime project ID of the
+   *     function and `us-central1` as location will be used:
+   *     `{channel-id}`
+   *
+   * If not specified, the default Firebase channel will be used:
+   * `projects/{project}/locations/us-central1/channels/firebase`
+   */
+  channel?: string;
+
+  /**
+   * Eventarc event exact match filter.
+   */
+  filters?: Record<string, string>;
+}
+
+export type CloudEventHandler = (event: CloudEvent<any>) => any | Promise<any>;
+
+/** Handle an Eventarc event published on the default channel. */
+export function onCustomEventPublished<T = any>(
+  eventType: string,
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>>;
+
+export function onCustomEventPublished<T = any>(
+  opts: EventarcTriggerOptions,
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>>;
+
+export function onCustomEventPublished<T = any>(
+  eventTypeOrOpts: string | EventarcTriggerOptions,
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>> {
+  let opts: EventarcTriggerOptions;
+  if (typeof eventTypeOrOpts === 'string') {
+    opts = {
+      eventType: eventTypeOrOpts as string,
+    };
+  } else if (typeof eventTypeOrOpts === 'object') {
+    opts = eventTypeOrOpts as EventarcTriggerOptions;
+  }
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(raw as CloudEvent<T>);
+  };
+
+  func.run = handler;
+
+  const channel = opts.channel ?? 'locations/us-central1/channels/firebase';
+
+  const baseOpts = options.optionsToEndpoint(options.getGlobalOptions());
+  const specificOpts = options.optionsToEndpoint(opts);
+
+  const endpoint: ManifestEndpoint = {
+    platform: 'gcfv2',
+    ...baseOpts,
+    ...specificOpts,
+    labels: {
+      ...baseOpts?.labels,
+      ...specificOpts?.labels,
+    },
+    eventTrigger: {
+      eventType: opts.eventType,
+      eventFilters: opts.filters,
+      retry: false,
+      channel,
+    },
+  };
+  if (!opts.filters) {
+    delete endpoint.eventTrigger.eventFilters;
```

Comment:
nit: why not just use `copyIfPresent` instead of inserting & deleting?

Reply:
Done.